ENIGMAsystem/SHELL/Bridges/General/Win32-Direct3D9.cpp

```diff
@@ -40,183 +39,184 @@ LPDIRECT3D9 d3dobj; // the pointer to our Direct3D interface
 
 namespace enigma {
 
+extern HWND hWnd;
+
 namespace dx9 {
 
 ContextManager* d3dmgr; // the pointer to the device class
 
 } // namespace dx9
 
-  extern bool forceSoftwareVertexProcessing;
-  bool Direct3D9Managed = true;
+extern HWND hWnd;
+extern bool forceSoftwareVertexProcessing;
+bool Direct3D9Managed = true;
 
-  void OnDeviceLost() {
-    d3dmgr->device->EndScene();
-    if (!Direct3D9Managed) return; // lost device only happens in managed d3d9
-    for (vector<Surface*>::iterator it = Surfaces.begin(); it != Surfaces.end(); it++) {
-      (*it)->OnDeviceLost();
-    }
+void OnDeviceLost() {
+  d3dmgr->device->EndScene();
+  if (!Direct3D9Managed) return; // lost device only happens in managed d3d9
+  for (vector<Surface*>::iterator it = Surfaces.begin(); it != Surfaces.end(); it++) {
+    (*it)->OnDeviceLost();
   }
+}
 
-  void OnDeviceReset() {
-    d3dmgr->device->BeginScene();
-    if (!Direct3D9Managed) return; // lost device only happens in managed d3d9
-    for (vector<Surface*>::iterator it = Surfaces.begin(); it != Surfaces.end(); it++) {
-      (*it)->OnDeviceReset();
-    }
+void OnDeviceReset() {
+  d3dmgr->device->BeginScene();
+  if (!Direct3D9Managed) return; // lost device only happens in managed d3d9
+  for (vector<Surface*>::iterator it = Surfaces.begin(); it != Surfaces.end(); it++) {
+    (*it)->OnDeviceReset();
   }
+}
 
-  void Reset(D3DPRESENT_PARAMETERS *d3dpp) {
-    OnDeviceLost();
-    HRESULT hr = d3dmgr->device->Reset(d3dpp);
-    if (FAILED(hr)) {
-      MessageBox(
-        enigma::hWnd,
-        TEXT("Device Reset Failed"), TEXT("Error"),
-        MB_ICONERROR | MB_OK
-      );
-      return;  // should probably force the game closed
-    }
-    // Reapply the stored render states and what not
-    d3dmgr->RestoreState();
-    OnDeviceReset();
-  }
+void Reset(D3DPRESENT_PARAMETERS *d3dpp) {
+  OnDeviceLost();
 
-  extern void (*WindowResizedCallback)();
-  void WindowResized() {
-    if (d3dmgr == NULL) { return; }
-    IDirect3DSwapChain9 *sc;
-    d3dmgr->device->GetSwapChain(0, &sc);
-    D3DPRESENT_PARAMETERS d3dpp;
-    sc->GetPresentParameters(&d3dpp);
-    int ww = enigma_user::window_get_width(),
-        wh = enigma_user::window_get_height();
-    d3dpp.BackBufferWidth = ww <= 0 ? 1 : ww;
-    d3dpp.BackBufferHeight = wh <= 0 ? 1 : wh;
-    sc->Release();
-    Reset(&d3dpp);
-
-    // clear the window color, viewport does not need set because backbuffer was just recreated
-    enigma_user::draw_clear(enigma_user::window_get_color());
+  HRESULT hr = d3dmgr->device->Reset(d3dpp);
+  if (FAILED(hr)) {
+    show_error("Device Reset Failed", true);
```

Comment:
I changed this from `MessageBox` to `show_error` so I could make the error fatal, which it is.